### DISCUSSION TITLE
Computed properties / mutators / accessors?

### DISCUSSION THREAD

**Discussion by mergims** [spatie/laravel-data#293] [General]
Is there any convenient way to implement accessors? Ideally I'd like to have private properties in a Data object, which can be accessed via an accessor `getPropertyName`. Also it would be useful to have computed properties that are included when normalizing or serializing a Data object.

**Sidenote:** i'm using `v1.5.1` since i'm locked on php 8.0 in my current project.

**Comment by rubenvanassche** [comment]
Hi @mergims,

We're not updating v1 anymore so expect no changes over there. As for computed properties and mutators, that's a path I'm currently not willing to take with the package due already a lot of complexity and you'll lose IDE completion. I guess it could be built with a trait but that something we're not going to provide at the moment.
